server/src/main/java/io/crate/execution/dsl/projection/builder/InputColumns.java

```diff
@@ -251,6 +251,9 @@ public Symbol visitReference(Reference ref, SourceSymbols sourceSymbols) {
         }
         InputColumn inputColumn = sourceSymbols.inputs.get(ref);
         if (inputColumn == null) {
+            if (ref.defaultExpression() != null) {
+                return ref.defaultExpression();
+            }
```

Comment:
Doesn't this have to be moved down below the subscriptOnRoot case to ensure we don't fallback to using a default expression on a select query?

Reply:
Honestly, I had it like that and then changed it to this. Do you have an example please?

Reply:
Does this subscript part require an array?

Reply:
Some of the test cases that fail if adding a throws clause in the "subscript works" case:


```
InsertAnalyzerTest. test_nested_primary_key_can_be_used_as_conflict_target_with_dotted_column_name
InsertAnalyzerTest. test_nested_primary_key_can_be_used_as_conflict_target_with_subscript_notation
InformationSchemaTest. testPartitionsNestedCol
InformationSchemaTest. testSelectPartitionedTablesFromInformationSchemaTable
InsertIntoIntegrationTest. testInsertIntoTablePartitionedOnGeneratedColumnBasedOnColumnWithinObject
InsertIntoIntegrationTest. testInsertIntoTableWithNestedPartitionedByFromQuery
InsertIntoIntegrationTest. testInsertIntoTableWithNestedPrimaryKeyFromQuery
PartitionedTableIntegrationTest. testPartitionedTableNestedAllConstraintsRoundTrip
PartitionedTableIntegrationTest. testPartitionedTableNestedPk
PartitionedTableIntegrationTest. testSelectPartitionValueFromInformationSchema
PartitionedTableIntegrationTest. test_insert_from_subquery_and_values_into_same_partition_for_object_ts_field_as_partition_key
PartitionedTableIntegrationTest. test_nested_partition_column_is_included_when_selecting_the_object_but_not_in_the_source
```

I think if there were a default clause there it could be a problem. Looks like we currently don't allow defaults on objects so not sure if you can create a case which already breaks, but I think it would break if we did

Reply:
Thx, I'll fix it, I also opened an issue for default values on arrays: https://github.com/crate/crate/issues/13148

Reply:
pushed (squashed the change)